### DISCUSSION TITLE
is it possible to set trust level for generated key or existing key using `rnpgp`?

### DISCUSSION THREAD

**Discussion by shemeshg** [rnpgp/rnp#2164] [Q&A]
See the `unknown` from `gpg --list-keys`
```
pub   rsa2048 2023-12-27 [SC] [expires: 2025-12-26]
      2FB12111E7B837E1A429F23EE96B81A18BC52564
uid           [ unknown] RSA (Encrypt or Sign) 2048-bit key <macos@localhost>
sub   rsa2048 2023-12-27 [E] [expires: 2025-12-26]
```

for created key using
```
./rnpkeys -g --homedir ~/.gnupg
```
*. can I self sign it using rnp c++ interface so it would be marked as `FULL`?
```
pub   rsa3072 2021-06-04 [SC]
      6A05080C049FCD4EDCBB3F611CA9424DDD85177F
uid           [  full  ] jacco@pastorias.com
```

I need it mostly for imported keys using
```
bool isSccussfull = import_keys(rnp.ffi, filePath, RNP_LOAD_SAVE_PUBLIC_KEYS);
```

in `gpgme` I used something like
```
gpgme_error_t err = gpgme_op_keysign(ctx, *key, NULL, 0, 0);
```

it is important so it will be possible to use it with `gopass`/`pass` without further key manipulation.

Thanks

**Comment by ni4** [comment]
GnuPG uses it's own key storage format, and stores trust information there. So it may be set only manually via the GnuPG command line.

**Comment by shemeshg** [comment]
Thanks.
